ext/BraketSimulatorPythonExt/BraketSimulatorPythonExt.jl

```diff
@@ -38,275 +42,37 @@ function __init__()
-    PythonCall.pyconvert_add_rule("braket.ir.jaqcd.results:Expectation", Braket.Result, jl_convert_expectation)
-    PythonCall.pyconvert_add_rule("braket.ir.jaqcd.results:Probability", Braket.Result, jl_convert_probability)
-    PythonCall.pyconvert_add_rule("braket.ir.jaqcd.results:Sample", Braket.Result, jl_convert_sample)
-    PythonCall.pyconvert_add_rule("braket.ir.jaqcd.results:StateVector", Braket.Result, jl_convert_statevector)
-    PythonCall.pyconvert_add_rule("braket.ir.jaqcd.results:DensityMatrix", Braket.Result, jl_convert_densitymatrix)
-    PythonCall.pyconvert_add_rule("braket.ir.jaqcd.results:Variance", Braket.Result, jl_convert_variance)
-    PythonCall.pyconvert_add_rule("braket.ir.jaqcd.results:AdjointGradient", Braket.Result, jl_convert_adjointgradient)
-    PythonCall.pyconvert_add_rule("braket.ir.openqasm.program_v1:Program", OpenQasmProgram, jl_convert_oqprogram)
-end
-
-function BraketSimulator.parse_program(simulator::D, program::OpenQasmProgram, shots::Int) where {D<:AbstractSimulator}
-    pc      = braket[].default_simulator.openqasm.program_context.ProgramContext()
-    interp  = braket[].default_simulator.openqasm.interpreter.Interpreter(pc)
-    inputs  = isnothing(program.inputs) ? pybuiltins.None : pydict(program.inputs)
-    py_circ = interp.build_circuit(source=pystr(program.source), inputs=inputs, is_file=endswith(program.source, ".qasm"))
-    if shots > 0
-        py_circ.instructions += py_circ.basis_rotation_instructions
-    end
-    circ = pyconvert(Circuit, py_circ)
-    @assert qubit_count(circ) <= properties(simulator).paradigm.qubitCount "parsed circuit's qubit count $(qubit_count(circ)) is larger than maximum ($(properties(simulator).paradigm.qubitCount)) for simulator type $D."
-    return ir(circ, Val(:JAQCD))
+    pyimport("braket.ir.openqasm.program_v1")
+    pyimport("braket.task_result.task_metadata_v1")
+    pyimport("braket.task_result.additional_metadata")
 end
 
-function simulate(
-    simulator::AbstractSimulator,
-    task_specs::Union{PyList{Any},NTuple{N,PyIterable}},
-    args...;
-    input::Union{PyList{Any},PyDict{Any,Any},Py}=PyDict{Any,Any}(),
-    kwargs...,
-) where {N}
-    # handle inputs
-    jl_inputs = nothing
-    shots = args[end]
-    stats = @timed begin
-        if input isa PyDict{Any,Any}
-            jl_inputs = pyconvert(Dict{String,Float64}, input)
-        else
-            jl_inputs = [pyconvert(Dict{String,Float64}, py_inputs) for py_inputs in input]
-        end
-        jl_specs = map(task_specs) do spec 
-            return if pyhasattr(spec, "source")
-                pyconvert(OpenQasmProgram, spec)
-            else
-                pyconvert(Program, spec)
+@recompile_invalidations begin
+    function simulate(simulator::AbstractSimulator,
+                      task_specs::PyList{Any},
+                      args...;
+                      max_parallel::Int=-1,
+                      inputs::Union{PyDict{Any, Any}, PyList{Any}, Nothing} = nothing,
```

Comment:
What is this parameter for? If it's the inputs to the OpenQASM program, those would have already been bound by the time this hit this simulator.

Reply:
It can be inputs to a JAQCD program too, and also would allow you to re-use the same `OpenQASMProgram` with many sets of inputs in the future.

Reply:
Ah, so extended functionality beyond what the SDK lets you do; would definitely be good to take advantage of this down the line